frontend/src/components/homepage/Homepage.js

```diff
@@ -1,13 +1,13 @@
-import CreatePost from "../CreatePost/CreatePost";
 import Feed from "../feed/Feed";
-import { useNavigate} from "react-router-dom";
-
-const Homepage = ({ navigate }) => {
+import { useContext } from 'react'
+import { UserContext } from '../../context/UserContext';
 
+const Homepage = (props) => {
+    const {userInfo} = useContext(UserContext);
 
     return (
         <>
-        <Feed navigate={ useNavigate() }/>
+            {userInfo ? <Feed navigate={props.navigate}/> : <div>You need an account to view and create posts.</div>}
```

Comment:
I may have to change this around a little bit, I was hoping to display a little login form somewhere on the page if the user is not logged in